optuna/visualization/_param_importances.py

```diff
@@ -122,56 +122,72 @@ def objective(trial):
         target:
             A function to specify the value to display. If it is :obj:`None` and ``study`` is being
             used for single-objective optimization, the objective values are plotted.
+            For multi-objective optimization, all objectives will be plotted if ``target``
+            is :obj:`None`.
 
             .. note::
-                Specify this argument if ``study`` is being used for multi-objective
-                optimization. For example, to get the hyperparameter importance of the first
-                objective, use ``target=lambda t: t.values[0]`` for the target parameter.
+                This argument can be used to specify which objective to plot if ``study``is being
+                used for multi-objective optimization. For example, to get only the hyperparameter
+                importance of the first objective, use ``target=lambda t: t.values[0]`` for the
+                target parameter.
         target_name:
-            Target's name to display on the axis label.
+            Target's name to display on the legend.
 
     Returns:
         A :class:`plotly.graph_objs.Figure` object.
     """
 
     _imports.check()
 
-    importances_info = _get_importances_info(study, evaluator, params, target, target_name)
-    hover_template = _get_hover_template(importances_info, study)
-    return _get_importances_plot(importances_info, hover_template)
+    if target or not study._is_multi_objective():
+        importances_infos: tuple[_ImportancesInfo, ...] = (
+            _get_importances_info(
+                study, evaluator, params, target=target, target_name=target_name
+            ),
+        )
+
+    else:
+        n_objectives = len(study.directions)
+        importances_infos = tuple(
+            _get_importances_info(
+                study,
+                evaluator,
+                params,
+                target=lambda t: t.values[objective_id],
+                target_name=f"{target_name} {objective_id}",
```

Comment:
Maybe something fancier here? Like modifying the API to allow users pass tuples of objective names, defaulting to  `Objective value {n}`? I'm very much open to the ideas.

Reply:
Sounds like a great feature to add. Could you please add it to the follow-up task list?

Reply:
I've rolled it over to a separate feature request - https://github.com/optuna/optuna/issues/4895.

Reply:
Thanks for the detailed explanation. Sound good.

Reply:
Thanks for a good idea!
Maybe we can take multiple `target`s as well.